sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2034,4 +2034,25 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       }
     }
   }
+
+  test("SPARK-21677: json_tuple throws NullPointException when column is null as string type") {
```

Comment:
Could we move this to `spark/sql/core/src/test/resources/sql-tests/inputs/json-functions.sql` and/or `spark/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/JsonExpressionsSuite.scala`?